### ISSUE TITLE
Non allowed categories names

### ISSUE THREAD

**Issue by riless** [NVIDIA/DIGITS#836]
I've noticed some issues when I try to build a dataset with categories named "_" (underscore) or " " (space).

Say I have this categories (folders of images): {A, B, C, D, _} for the training set, and the same thing for the validation set.

Digits parses all the catégories for train (Number if Categories:5), but ignores the '_' category in the val (Number of Categories:4)

It's is not a blocking bug as I can rename the category '_' and everything will work fine. But adding a little section in Digits where the non allowed categories names are indicated can help a lot.

**Comment by gheinrich** [comment]
Thanks for the bug report. I think the issue is that underscores are replaced with spaces when parsing folders [there](https://github.com/NVIDIA/DIGITS/blob/v4.0.0-rc.1/tools/parse_folder.py#L358) and if your label consists of just an underscore then the label is set to a single space which won't be recognized as a non-empty line after stripping in [get_labels()](https://github.com/NVIDIA/DIGITS/blob/v4.0.0-rc.1/digits/dataset/tasks/create_db.py#L278-L282). 

That causes some issues in DIGITS, e.g. you can't explore your databases and inference will probably fail intermittently. However that doesn't mean your `_` samples won't be included in the training and validation sets and I think you'll see there in your `train.txt` and `val.txt` files. Your model will use them during training.

Admittedly `_` is a pretty odd label but I think there is indeed a bug in DIGITS.

**Comment by riless** [comment]
Thank you for your explanations.

The samples are just not included in the val.txt, so they are not used in the train process.

**Comment by gheinrich** [comment]
> The samples are just not included in the val.txt

That is odd. Is it possible that you have so few samples in this category that none were picked  when randomly creating the validation set? Either way, `val.txt` is used for validation, not training.

**Comment by mizadyya** [comment]
I guess this is an issue with Caffe. If the name of an image also has '_'  that image will not be included in the LMDB file and so on. Any way I needed to rename all those images in order to solve this issue.

**Comment by gheinrich** [comment]
I don't think Caffe is to blame here as the LMDB creation is happening in DIGITS. You can use underscores in combination with other standard characters but naming a file just `_` is looking for troubles :-)

**Comment by mizadyya** [comment]
I faced this problem before using DIGITS (I was using python for creating LMDB files).  The file names that I was working on "included" _ rather  being just _. Simply it ignores files that their name has _ char. 
So there's a trouble and it's with Caffe  and or  DIGITS. :-)

**Comment by gheinrich** [comment]
Hi @mizadyya I find that hard to believe given that the file name isn't featured in LMDB. The LMDB just has key/value pairs. Whatever problem you had I think was due to the script you used to create the LMDB.

There is indeed an issue in DIGITS when the label name is exactly `_`. Other than that, you should be fine with underscores.

**Comment by gheinrich** [comment]
Closing due to inactivity.
